akka-persistence/src/main/scala/akka/persistence/Persistence.scala

```diff
@@ -105,8 +105,18 @@ trait PersistenceRecovery {
    * To skip recovery completely return `Recovery.none`.
    */
   def recovery: Recovery = Recovery()
-  //#persistence-recovery
 }
+//#persistence-recovery
+
+//#persistence-stash
+trait PersistenceStash extends Stash with StashFactory {
+  /**
+   * The returned [[StashOverflowStrategy]] object determine how to handle the received message when the
+   * size of Stash exceeds the capacity of the Stash.
+   */
+  def stashOverflowStrategy: StashOverflowStrategy = ThrowOverflowExceptionStrategy
```

Comment:
I'm sorry for late (china new year). The words in the docs is short of the relation between stash and mailbox, eg. the deffience between inherited user's ```unstash()``` and ```unstashAll()```. IMO, there is a real needed to make some words in the docs for a new stash section. If you don't mind, I will commit some words in this pr in recent days.